lib/add_magic_comment.rb

```diff
@@ -16,11 +16,20 @@ module AddMagicComment
 
   def self.process(argv)
     directory = argv.first || Dir.pwd
+    ignore_pattern = Regexp.new(argv[1]) rescue nil
+    # Allow providing as either a Regex or a String
+    if ignore_pattern && ignore_pattern.is_a?(String)
```

Comment:
To rephrase my first question:  is there every a case where this can be true?

`ignore_pattern && ignore_pattern.is_a?(String)`

My read of the code is that `ignore_pattern` will always be a `Regexp` or `nil`.